rust/publish-ghost/src/lib.rs

```diff
@@ -76,6 +77,26 @@ pub struct Cli {
     #[arg(long, conflicts_with = "push")]
     pull: bool,
 
+    // Push as draft
+    #[arg(
+        long,
+        group = "publish_type",
+        requires = "push",
+        default_value_t = true
+    )]
+    draft: bool,
+
+    // Publish pushed, page or post
+    #[arg(long, group = "publish_type", requires = "push")]
+    publish: bool,
+
+    // schedule pushed, page or post
+    #[arg(long, group = "publish_type", requires = "push")]
+    schedule: bool,
+
+    #[arg(long, requires = "schedule")]
+    schedule_date: Option<DateTime<Utc>>,
```

Comment:
I seems that you can't use `--schedule` without a time so why not just have one arg here:

```suggestion
    /// Schedule a page or post to be published at a date/time
    #[arg(long, requires = "push", conflicts_with = "draft", conflicts_with = "publish", value_parser = parse_date_time)]
    schedule: Option<DateTime<Utc>>,
```

Reply:
yeah, that's better